frontend/packages/client/src/components/WalletConnect.js

```diff
@@ -1,46 +1,188 @@
+                    height: '40px',
+                  }
+                : { width: '119px', height: '32px' }
+            }
+          >
+            {loggedIn ? (
+              <div className={containerButtonStyle}>
+                <Blockies
+                  seed={addr}
+                  size={notMobile ? 6.5 : 5}
+                  scale={4}
+                  className="blockies"
+                />
+                <div className={containerAddressStyle}>
+                  <p className={addressStyle}>{truncateAddress(addr, 4, 4)}</p>
+                </div>
+              </div>
+            ) : (
+              <>
+                <span>Connect</span>
+                <span className="is-hidden-mobile">&nbsp;Wallet</span>
+              </>
+            )}
+          </button>
+        </div>
+
+        <div
+          className="dropdown-menu wallet-connect-content"
+          id="dropdown-menu"
+          role="menu"
+          ref={dropdownRef}
+          style={!notMobile ? { left: '-160px' } : {}}
+        >
+          <div className="dropdown-content p-0" style={{ width: '277px' }}>
+            <div className="px-4 pt-4 pb-2">
+              <Tooltip
+                classNames="is-flex is-flex-grow-1 is-align-items-center transition-all"
+                position="top"
+                text="Copied!"
+                alwaysVisible={true}
+                enabled={addressCopied}
+              >
+                <CopyToClipboard text={addr} onCopy={markAddressCopied}>
+                  <div
+                    className="columns flex-1 is-mobile m-0 px-4 py-0 rounded-sm button is-white border-light"
+                    style={{
+                      borderColor: 'hsl(0deg, 0%, 86%)',
+                      height: '32px',
+                    }}
+                  >
+                    <div className="column p-0 is-flex is-align-items-center">
+                      <span className="small-text">{addr}</span>
+                    </div>
+                    <div className="column p-0 is-flex is-align-items-center is-narrow">
+                      <div
+                        className="is-flex is-align-items-center py-0 px-1"
+                        style={{ height: '23px' }}
+                      >
+                        <Copy />
+                      </div>
+                    </div>
+                  </div>
+                </CopyToClipboard>
+              </Tooltip>
+            </div>
+
+            <hr className="dropdown-divider" />
+            <div className="px-4 pb-4 pt-2">
+              <div
+                className="button is-fullwidth rounded-sm is-uppercase is-flex small-text has-text-white has-background-black"
+                style={{ height: '32px' }}
+                onClick={signOut}
+              >
+                disconnect
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </>
```

Comment:
there's a upcoming update for this component, I thought about making it more modular but decided to wait for the update and see how it can be split into more components